### ISSUE TITLE
FATAL {crash} occurs on a simple query

### ISSUE THREAD

**Issue by wajda** [arangodb/arangodb#15476]
## My Environment

* __ArangoDB Version__:        3.8.4-1
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Infrastructure__:               localhost
* __Operating System__:        Fedora 35
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        SSD
* __Used Package__:           RPM (from https://download.arangodb.com/arangodb38/RPM)

## Component, Query & Data

__AQL query__:
```aql
FOR x IN foo
    COLLECT WITH COUNT INTO cnt
    RETURN 1
```
`foo` can be any (even empty) collection.

__AQL explain and/or profile (if applicable)__:

__Dataset__: doesn't matter

## Steps to reproduce

1. Run the above query against any arbitrary collection (including empty one)

__Problem__:

The above query makes the server to crash.
**Note:** with `RETURN cnt` it works as expected. It's returning a constant value makes it crashing, 

```
2022-01-07T09:45:49Z [74805] FATAL [a7902] {crash} 💥 ArangoDB 3.8.4 [linux], thread 4 [SchedWorker] caught unexpected signal 11 (SIGSEGV) accessing address 0x0000000000000010: signal handler invoked - image base address: 0x00007fd684f2e000 - CPU context: rip: 0x00007fd686109ade, rsp: 0x00007fd67dffbe50, efl: 0x0000000000010206, rbp: 0x00007fd67dffc4e0, rsi: 0x0000000000000002, rdi: 0x0000000000000001, rax: 0x00007fd4f98231a0, rbx: 0x00007fd67dffc940, rcx: 0x00007fd664442308, rdx: 0x0000000000000001, r8: 0x0000000000000001, r9: 0x00007fd664442310, r10: 0x0000000000000018, r11: 0x00007fd67dfffb20, r12: 0x0000000000000000, r13: 0x00007fd664442300, r14: 0x0000000000000000, r15: 0x00007fd67dffc5b8
2022-01-07T09:45:49Z [74805] INFO [c962b] {crash} Backtrace of thread 4 [SchedWorker]
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  1 [+0x0000000000806fd5] (anonymous namespace)::crashHandlerSignalHandler(int, siginfo_t*, void*) (+0x35)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  2 [+0x00000000039e34d3] sigprocmask (+0x21)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  3 [+0x00000000013c0099] SingleServerQueryInstanciator::after(arangodb::aql::ExecutionNode*) (+0x119)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  4 [+0x0000000001399824] arangodb::aql::ExecutionNode::walk(arangodb::aql::WalkerWorkerBase<arangodb::aql::ExecutionNode>&) (+0x134)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  5 [+0x00000000013bd990] arangodb::aql::ExecutionEngine::instantiateFromPlan(arangodb::aql::Query&, arangodb::aql::ExecutionPlan&, bool, arangodb::aql::SerializationFormat) (+0x2f0)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  6 [+0x00000000006afabf] arangodb::aql::Query::prepareQuery(arangodb::aql::SerializationFormat) [clone .constprop.0] (+0x7bf)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  7 [+0x00000000015ec775] arangodb::aql::Query::execute(arangodb::aql::QueryResult&) (+0xaa5)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  8 [+0x00000000006d0e1c] arangodb::RestCursorHandler::processQuery(bool) [clone .constprop.0] (+0x2c)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame  9 [+0x0000000001973a4e] arangodb::RestCursorHandler::registerQueryOrCursor(arangodb::velocypack::Slice const&) (+0x6fe)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 10 [+0x000000000197476b] arangodb::RestCursorHandler::execute() (+0x8fb)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 11 [+0x00000000019e3286] arangodb::rest::RestHandler::executeEngine(bool) (+0xb6)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 12 [+0x0000000001a0d26a] arangodb::rest::RestHandler::runHandlerStateMachine() (+0x36a)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 13 [+0x000000000169535b] arangodb::Scheduler::WorkItem<arangodb::rest::CommTask::handleRequestAsync(std::shared_ptr<arangodb::rest::RestHandler>, unsigned long*)::{lambda()#1}>::invoke() (+0xfb)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 14 [+0x000000000186cc5f] arangodb::SupervisedScheduler::runWorker() (+0x41f)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 15 [+0x0000000000814039] arangodb::Thread::startThread(void*) (+0x69)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 16 [+0x00000000007b5a4f] ThreadStarter(void*) [clone .lto_priv.0] (+0x8f)
2022-01-07T09:45:49Z [74805] INFO [308c3] {crash} frame 17 [+0x00000000039ea2a7] start (+0x70)
2022-01-07T09:45:49Z [74805] INFO [ded81] {crash} available physical memory: 33252749312, rss usage: 1129127936, vsz usage: 6865514496, threads: 51

```

__Expected result__:

```
[
  1
]
```

**Comment by jsteemann** [comment]
Reproduce case:
```js
db._create("foo");
q = `FOR x IN foo COLLECT WITH COUNT INTO cnt RETURN 1`; 
db._query(q);
```
Produces:
```
2022-01-07T10:00:08Z [22657] FATAL [a7902] {crash} 💥 ArangoDB 3.8.5 enterprise [linux], thread 21 [Console] caught unexpected signal 6 (SIGABRT): assertion failed in arangod/Aql/CollectNode.cpp:359 [createBlock]: aggregateVariables().size() == 1 - image base address: 0x000055e328e12000
2022-01-07T10:00:08Z [22657] INFO [c962b] {crash} Backtrace of thread 21 [Console]
2022-01-07T10:00:08Z [22657] INFO [308c3] {crash} frame  1 [+0x000000000231d67b] arangodb::CrashHandler::crash(char const*) (+0x1b)
2022-01-07T10:00:08Z [22657] INFO [308c3] {crash} frame  2 [+0x000000000231d881] arangodb::CrashHandler::assertionFailure(char const*, int, char const*, char const*) (+0x111)
2022-01-07T10:00:08Z [22657] INFO [308c3] {crash} frame  3 [+0x0000000000fff2c3] arangodb::aql::CollectNode::createBlock(arangodb::aql::ExecutionEngine&, std::unordered_map<arangodb::aql::ExecutionNode*, arangodb::aql::ExecutionBlock*, std::hash<arangodb::aql::ExecutionNode*>, std::equal_to<arangodb::aql::ExecutionNode*>, std::allocator<std::pair<arangodb::aql::ExecutionNode* const, arangodb::aql::ExecutionBlock*> > > const&) const (+0x2f53)
```

**Comment by jsteemann** [comment]
Bugfix PR for 3.8: https://github.com/arangodb/arangodb/pull/15478
The fix will become available with 3.8.5
